pkg/sensors/tracing/generickprobe.go

```diff
@@ -290,9 +290,13 @@ func preValidateKprobes(name string, kprobes []v1alpha1.KProbeSpec) error {
 	for i := range kprobes {
 		f := &kprobes[i]
 
-		hasOverride := selectors.HasOverride(f)
-		if hasOverride && !bpf.HasOverrideHelper() {
-			return fmt.Errorf("Error override action not supported, bpf_override_return helper not available")
+		if selectors.HasOverride(f) {
+			if !bpf.HasOverrideHelper() {
+				return fmt.Errorf("Error override action not supported, bpf_override_return helper not available")
+			}
+			if !f.Syscall {
+				return fmt.Errorf("Error override action can be used only with syscalls")
+			}
```

Comment:
just a question: why did we put this limitation? Because it's too easy to shoot yourself in the foot and crash the kernel with an override on any other kernel function?

Reply:
the override is limited for syscalls and few other functions (marked with ALLOW_ERROR_INJECTION in kernel)
so we limited this for syscalls only, because I don't think we have a usecase for the rest

Reply:
ah I see thanks! :)